designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ScenarioActivityApiEndpoints.scala

```diff
@@ -0,0 +1,280 @@
+import sttp.model.StatusCode.{InternalServerError, NotFound, Ok}
+import sttp.model.{HeaderNames, MediaType}
+import sttp.tapir.EndpointIO.Example
+import sttp.tapir._
+import sttp.tapir.derevo.schema
+import sttp.tapir.json.circe.jsonBody
+
+import java.io.InputStream
+import java.time.Instant
+
+class ScenarioActivityApiEndpoints(auth: EndpointInput[AuthCredentials]) extends BaseEndpointDefinitions {
+
+  import ScenarioActivityApiEndpoints.Dtos.ScenarioActivityErrors._
+  import ScenarioActivityApiEndpoints.Dtos._
+  import TapirCodecs.ContentDispositionCodec._
+  import TapirCodecs.HeaderCodec._
+  import TapirCodecs.ScenarioNameCodec._
+  import TapirCodecs.VersionIdCodec._
+
+  lazy val scenarioActivityEndpoint: SecuredEndpoint[ProcessName, ScenarioActivityErrors, ScenarioActivity, Any] =
+    baseNuApiEndpoint
+      .summary("Scenario activity service")
+      .tag("Scenario")
+      .get
+      .in("processes" / path[ProcessName]("scenarioName") / "activity")
+      .out(
+        statusCode(Ok).and(
+          jsonBody[ScenarioActivity].example(
+            Example.of(
+              summary = Some("Display scenario activity"),
+              value = ScenarioActivity(
+                comments = List(
+                  Comment(
+                    id = 1L,
+                    processVersionId = 1L,
+                    content = "some comment",
+                    user = "test",
+                    createDate = Instant.parse("2024-01-17T14:21:17Z")
+                  )
+                ),
+                attachments = List(
+                  Attachment(
+                    id = 1L,
+                    processVersionId = 1L,
+                    fileName = "some_file.txt",
+                    user = "test",
+                    createDate = Instant.parse("2024-01-17T14:21:17Z")
+                  )
+                )
+              )
+            )
+          )
+        )
+      )
+      .errorOut(scenarioNotFoundErrorOutput)
+      .withSecurity(auth)
+
+  lazy val addCommentEndpoint: SecuredEndpoint[AddCommentRequest, ScenarioActivityErrors, Unit, Any] =
+    baseNuApiEndpoint
+      .summary("Add scenario comment service")
+      .tag("Scenario")
+      .post
+      .in(
+        ("processes" / path[ProcessName]("scenarioName") / path[VersionId]("versionId") / "activity"
+          / "comments" / stringBody).mapTo[AddCommentRequest]
+      )
+      .out(statusCode(Ok))
+      .errorOut(scenarioNotFoundErrorOutput)
+      .withSecurity(auth)
+
+  lazy val deleteCommentEndpoint: SecuredEndpoint[DeleteCommentRequest, ScenarioActivityErrors, Unit, Any] =
+    baseNuApiEndpoint
+      .summary("Delete process comment service")
+      .tag("Scenario")
+      .delete
+      .in(
+        ("processes" / path[ProcessName]("scenarioName") / "activity" / "comments"
+          / path[Long]("commentId")).mapTo[DeleteCommentRequest]
+      )
+      .out(statusCode(Ok))
+      .errorOut(
+        oneOf[ScenarioActivityErrors](
```

Comment:
There is one more fundamental problem with sharing the same error between these endpoints. E.g. in this endpoint, only the two of them are used (there is no `NoPermission` usage). When in the future, the service logic changes and it returns `NoPermission`, there won't be a compiler error that reminds you to handle this new, specific case. But it seems the Tapir approach is also limited according to this. 

We could have errors hierarchy per endpoint obliviously. But it doesn't solve the mentioned problem too. 

So, I don't expect you to improve it, but I wanted to write it to make sure we are aware of the problem.